e2e/pod_status.go

```diff
@@ -49,6 +51,43 @@ func isPodGone(cs *kubernetes.Clientset, podName, namespace string) wait.Conditi
 	}
 }
 
+func isReplicaSetSteady(cs *kubernetes.Clientset, replicaSetName, namespace, label string) wait.ConditionFunc {
+	return func() (bool, error) {
+		podList, err := ListPods(cs, namespace, label)
+		if err != nil {
+			return false, err
+		}
+
+		replicaSet, err := cs.AppsV1().ReplicaSets(namespace).Get(context.Background(), replicaSetName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		// check two things:
+		// 1. number of pods that are ready should equal that of spec
+		// 2. number of pods matching replicaSet's selector should equal that of spec
+		//    (in 0 replicas case, replicas should finish terminating before this comes true)
+		if replicaSet.Status.ReadyReplicas == (*replicaSet.Spec.Replicas) && int32(len(podList.Items)) == (*replicaSet.Spec.Replicas) {
```

Comment:
please export this function into an helper. `isReplicaSetSynchronized(replicaSet, podList)` or something like that.